### PR TITLE
Remove methods that no longer exist in native SDKs

### PR DESCRIPTION
**Pull request by nsingh-branch**
## Summary
Removed various deprecated methods and updates imports to use the latest Branch native SDKs.

## Motivation
Updating to the latest native SDKs required removing methods from the plugin as well.

## Type Of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing Instructions

Try out the `6.0.0-alpha.6` package in a Cordova test app on Android and iOS. Everything should build and run fine.

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

**Review by gdeluna-branch — APPROVED** [review]

File: src/ios/BranchSDK.m [hunk 1]
```diff
@@ -323,20 +311,17 @@ - (void)createBranchUniversalObject:(CDVInvokedUrlCommand*)command
 
   for (id key in properties) {
     if ([key isEqualToString:@"contentMetadata"]){
-      NSDictionary *metadata = (NSDictionary *)[properties valueForKey:key];
```
Comment (echo-branch) [inline src/ios/BranchSDK.m:326]: I'd like to see this part tested in a either the testbed or your own test app. The slight change in logic looks safe, but really want to be sure the command actually has the right data type in it.

File: src/ios/BranchSDK.m [hunk 2]
```diff
@@ -323,20 +311,17 @@ - (void)createBranchUniversalObject:(CDVInvokedUrlCommand*)command
...
-        [branchUniversalObj addMetadataKey:key_ value:[metadata valueForKey:key_]];
-      }
+        NSMutableDictionary<NSString *,NSString *> *metadata = (NSMutableDictionary<NSString *,NSString *> *)[properties valueForKey:key];
+        [[branchUniversalObj contentMetadata] setCustomMetadata:metadata];
     }
     else if ([key isEqualToString:@"contentIndexingMode"]) {
       NSString *indexingMode = [properties valueForKey:key];
       // Default contentIndexMode is always public
       if ([indexingMode isEqualToString:@"private"]) {
-        branchUniversalObj.contentIndexMode = BranchContentIndexModePrivate;
```
Comment (echo-branch) [inline src/ios/BranchSDK.m:336]: Double check that this is actually a one to one replacement behavior. Cause if it's not we might need to change the key check as well.

File: src/ios/BranchSDK.m (hunk 1 above)
Comment (nsingh-branch) [reply src/ios/BranchSDK.m:326]: Confirmed that setting contentMetadata still works and appears properly in requests.

File: src/ios/BranchSDK.m (hunk 2 above)
Comment (nsingh-branch) [reply src/ios/BranchSDK.m:336]: Yeah, based on my test setting to public/private properly adjusts the `publicly_indexable` field in requests.

**Pull request merged**